src/devices/Qmc5883l/Qmc5883l.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Numerics;
+using UnitsNet;
+
+namespace Iot.Device.Qmc5883l
+{
+    /// <summary>
+    /// 3-Axis Digital Compass QMC5883L
+    /// </summary>
+    [Interface("3-Axis Digital Compass QMC5883L")]
+    public class Qmc5883l : IDisposable
+    {
+        /// <summary>
+        /// QMC5883L Default I2C Address
+        /// </summary>
+        public const byte DefaultI2cAddress = 0x0D;
+        private I2cDevice _i2cDevice;
+
+        /// <summary>
+        /// QMC5883L Direction Vector
+        /// </summary>
+        [Telemetry]
+        public Vector3 Direction => ReadDirectionVector();
+
+        /// <summary>
+        /// QMC5883L Heading (DEG)
+        /// </summary>
+        public Angle Heading => VectorExtentsion.GetHeading(ReadDirectionVector());
```

Comment:
I would consider changing these to methods. Properties should not be expensive or return different values if called multiple times. You might even want to change them to the Try... pattern. So this would be `bool TryGetHeading(out Angle heading);`